### PR TITLE
If GallerySrc only contains 1 image, it now won't scroll through it and won't show the counter at top right

### PR DESCRIPTION
**Pull request by jonluca**

### CODE REVIEW

**Comment by jonluca** [comment]
This fixes a few bugs. 

If it's an album link on imgur that only contains 1 image, it won't show the counter in the top right, and if you scroll down it won't just "cycle" through that 1 image.

File: js/hoverzoom.js [hunk 1]
```diff
@@ -563,7 +563,7 @@ var hoverZoom = {
                 }
                 if (linkData.hoverZoomGallerySrc) {
                     var info = '';
-                    if (linkData.hoverZoomGallerySrc.length > 0) {
+                    if (linkData.hoverZoomGallerySrc.length > 1) {
```
Comment (extesy) [inline js/hoverzoom.js]: Here you are still applying hzGalleryInfoCss and creating hzGallery div.

File: js/hoverzoom.js [hunk 2]
```diff
@@ -945,6 +945,9 @@ var hoverZoom = {
             if (imgFullSize) {
                 var link = hz.currentLink, data = link.data();
                 if (data.hoverZoomGallerySrc) {
+                    if(data.hoverZoomGallerySrc.length === 1){
```
Comment (extesy) [inline js/hoverzoom.js]: Should merge this "if" with the previous "if" and also respect coding style (whitspaces).

File: manifest.json [hunk 3]
```diff
@@ -1,7 +1,7 @@
 {
     "name": "__MSG_extName__",
     "short_name": "__MSG_extShortName__",
-    "version": "1.0.76",
+    "version": "1.0.77",
```
Comment (extesy) [inline manifest.json]: 1.0.78

File: plugins/imgur_a.js [hunk 4]
```diff
@@ -29,6 +29,10 @@ hoverZoomPlugins.push({
 
             if (options.zoomVideos && (href.substr(-3) == 'gif' || href.substr(-4) == 'gifv')) {
                 data.hoverZoomSrc = [href.replace(/\.gifv?/, '.webm'), href.replace(/\.gifv?/, '.mp4'), href];
+                //future code for when imgur vids can be served over https
```
Comment (extesy) [inline plugins/imgur_a.js]: Remove this commented block. Https support should be done differently anyways.

File: plugins/imgur_a.js [hunk 5]
```diff
@@ -42,32 +46,41 @@ hoverZoomPlugins.push({
                     
                     switch (view) {
                         case 'signin':
-                            return;
+                        return;
```
Comment (extesy) [inline plugins/imgur_a.js]: restore identation

File: plugins/imgur_a.js [hunk 6]
```diff
@@ -42,32 +46,41 @@ hoverZoomPlugins.push({
...
 
                             // Future alternative: https://imgur.com/ajaxalbums/getimages/{hash}/hit.json?all=true
                             var albumUrl = 'https://api.imgur.com/3/album/' + hash + '.json';
                             $.ajax(albumUrl, {headers: {"Authorization": "Client-ID 1d8d9b36339e0e2"}}).done(function (imgur) {
                                 if (imgur.error) {
                                     data.hoverZoomSrc = createUrls(hash);
                                     res.push(link);
-                                } else {
+                                } 
+                                //todo: Possible faster implementation later on for single images - leave commented for now
```
Comment (extesy) [inline plugins/imgur_a.js]: Remove this commented out code. Gallery optimization should be done in common code, not in plugins.

File: plugins/imgur_a.js [hunk 7]
```diff
@@ -42,32 +46,41 @@ hoverZoomPlugins.push({
                     
                     switch (view) {
                         case 'signin':
-                            return;
+                        return;
                         case 'a': // album view:
                         case 'gallery':
-                            var anchor = matches[3];
```
Comment (extesy) [inline plugins/imgur_a.js]: Why did you move this block?

File: plugins/imgur_a.js [hunk 8]
```diff
@@ -42,32 +46,41 @@ hoverZoomPlugins.push({
...
+                                */
+                                else {
+                                    var anchor = matches[3];
+                                    data.hoverZoomGallerySrc = [];
+                                    data.hoverZoomGalleryCaption = [];
                                     imgur.data.images.forEach(function (img, index) {
                                         var urls = [img.link],
-                                            caption = img.title,
-                                            alreadyAdded = false;
+                                        caption = img.title,
```
Comment (extesy) [inline plugins/imgur_a.js]: restore identation

File: plugins/imgur_a.js [hunk 9]
```diff
@@ -42,32 +46,41 @@ hoverZoomPlugins.push({
...
+                                        alreadyAdded = false;
                                         for (var i=0, l=data.hoverZoomGallerySrc.length; i<l; i++) {
                                             if (data.hoverZoomGallerySrc[i].indexOf(urls[0]) != -1) {
                                                 alreadyAdded = true;
                                                 break;
                                             }
                                         }
                                         if (!alreadyAdded) {
-                                            if (caption != '' && img.description != '') {
+                                            if (caption != '' && caption != 'null' && img.description != '') {
```
Comment (extesy) [inline plugins/imgur_a.js]: if caption is string "null", shouldn't you clear it first before appending description? otherwise you'll get text "nullDescription"

File: plugins/imgur_a.js [hunk 10]
```diff
@@ -94,10 +107,10 @@ hoverZoomPlugins.push({
                                 }
                             });
                             break;
-                        case undefined:
+                            case undefined:
```
Comment (extesy) [inline plugins/imgur_a.js]: restore identation

File: plugins/imgur_a.js [hunk 11]
```diff
@@ -94,10 +107,10 @@ hoverZoomPlugins.push({
...
                         default: // single pic view
-                            data.hoverZoomSrc = createUrls(hash);
-                            res.push(link);
+                        data.hoverZoomSrc = createUrls(hash);
```
Comment (extesy) [inline plugins/imgur_a.js]: restore identation

**Comment by jonluca** [comment]
Alright, feel free to close this pull request and I'll make another one. I used a reindent program that messed up the whitespace, it should be fixed in my latest commit though. I'll just make a new pr when you close this one. Thanks for the suggestions/help!
